### PR TITLE
chore(dashboard): Integrate dashboard app into the SPA bundle

### PR DESCRIPTION
**Pull request by suddjian**
### SUMMARY

- Move Dashboard app's individual reducer and app initialization logic to the Single Page App `/views/App`
- Move remaining page-specific logic from the backend to the frontend (css, title)
- Convert the links in the dashboard list from anchors to react-router links

### TEST PLAN

- Run the changes in ephemeral env, navigate to dashboards, edit, reload, create new dashboards, go between dashboard page and other views, try url parameters such as `standalone`.
- Add a e2e test that navigates between pages

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -1853,11 +1853,12 @@ def dashboard(  # pylint: disable=too-many-locals
         }
 
         return self.render_template(
-            "superset/dashboard.html",
-            entry="dashboard",
+            "superset/crud_views.html",
```
Comment (nytai) [inline superset/views/core.py:1856]: We should, eventually, rename this template to `spa.html`

Comment (suddjian) [reply superset/views/core.py:1856]: seems like now could be the time

File: superset-frontend/src/components/Menu/Menu.tsx [hunk 2]
```diff
@@ -159,6 +162,10 @@ export function Menu({
 }: MenuProps) {
   const [dropdownOpen, setDropdownOpen] = useState(false);
 
+  // would useQueryParam here but not all apps provide a router context
+  const standalone = getUrlParam('standalone', 'boolean');
+  if (standalone) return <></>;
```
Comment (nytai) [inline superset-frontend/src/components/Menu/Menu.tsx:167]: oh this is cool, we'll support `standalone=true` for all pages

Comment (suddjian) [reply superset-frontend/src/components/Menu/Menu.tsx:167]: In the course of writing this I also learned about `standalone=2` which removes the dashboard header lol

Comment (suddjian) [reply superset-frontend/src/components/Menu/Menu.tsx:167]: we could have `standalone` be a set of bit flags where each bit is a different feature to remove

Comment (nytai) [reply superset-frontend/src/components/Menu/Menu.tsx:167]: 🤦

File: superset-frontend/src/components/ListViewCard/index.tsx [hunk 3]
```diff
@@ -169,13 +182,14 @@ function ListViewCard({
   imgPosition = 'top',
   cover,
 }: CardProps) {
+  const Link = url && linkComponent ? linkComponent : AnchorLink;
```
Comment (nytai) [inline superset-frontend/src/components/ListViewCard/index.tsx:185]: Is this just in case this is used outside of a react-router context?

Comment (suddjian) [reply superset-frontend/src/components/ListViewCard/index.tsx:185]: Most list view links still need to be anchor links. It's just the dashboard ones that can be `<Link>`s for now.

**Review by nytai — APPROVED** [review]

File: superset-frontend/src/components/ListViewCard/index.tsx [hunk 4]
```diff
@@ -169,13 +182,14 @@ function ListViewCard({
...
   return (
     <StyledCard
       data-test="styled-card"
       cover={
         cover || (
           <Cover>
-            <a href={url}>
+            <Link to={url!}>
```
Comment (rusackas) [inline superset-frontend/src/components/ListViewCard/index.tsx:192]: Maybe we can adda. test to make sure this renders as expected with/without the url?

File: superset-frontend/src/dashboard/App.jsx [hunk 5]
```diff
@@ -1,52 +0,0 @@
-/**
```
Comment (rusackas) [inline superset-frontend/src/dashboard/App.jsx:1]: 👏🥳

File: superset-frontend/src/views/CRUD/dashboard/DashboardCard.tsx [hunk 6]
```diff
@@ -139,7 +141,7 @@ function DashboardCard({
     <CardStyles
       onClick={() => {
         if (!bulkSelectEnabled) {
-          window.location.href = dashboard.url;
+          history.push(dashboard.url);
```
Comment (rusackas) [inline superset-frontend/src/views/CRUD/dashboard/DashboardCard.tsx:144]: Let's do an E2E test to go into and out of a dashboard, to make sure we can get hither and yon.

Comment (nytai) [reply superset-frontend/src/views/CRUD/dashboard/DashboardCard.tsx:144]: Would be good to have one that hits the dashboard url directly and one that navigates to it via the list view link

**Review by rusackas — APPROVED** [review]
Noted a couple opportunities for tests, but in general LGTM!!

**Pull request merged**